pkg/upgrade/update.go

```diff
@@ -56,74 +56,50 @@ func (u *simpleUpgrader) Update(ctx context.Context, cs, oldCs *api.OpenShiftMan
 		return err
 	}
 
-	if isScaleOp {
-		for _, agent := range cs.Properties.AgentPoolProfiles {
-			vms, err := listVMs(ctx, cs, vmc, agent.Role)
-			if err != nil {
-				return err
-			}
+	for _, agent := range cs.Properties.AgentPoolProfiles {
+		vms, err := listVMs(ctx, cs, vmc, agent.Role)
+		if err != nil {
+			return err
+		}
 
-			// wait for newly created VMs to reach readiness
-			for _, vm := range vms {
-				hostname := *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName
-				if _, found := vmsBefore[hostname]; !found {
-					log.Infof("waiting for %s to be ready", hostname)
-					err = u.WaitForReady(ctx, cs, agent.Role, hostname)
-					if err != nil {
-						return err
-					}
+		// wait for newly created VMs to reach readiness
+		for _, vm := range vms {
+			hostname := *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName
+			if _, found := vmsBefore[hostname]; !found {
+				log.Infof("waiting for %s to be ready", hostname)
+				err = u.WaitForReady(ctx, cs, agent.Role, hostname)
+				if err != nil {
+					return err
 				}
 			}
 		}
-
-		return nil
 	}
 
-	err = u.updateInPlace(ctx, cs, ssc, vmc, api.AgentPoolProfileRoleMaster)
-	if err != nil {
-		return err
-	}
+	// For PP day 1, scale is permitted but not any other sort of update.  When
```

Comment:
Why not make this configurable, at least to start soaking the upgrade plugin in tests?

Reply:
Sure, happy to do that.